docs/cloud-nativeness/opentelemetry.md

```diff
@@ -138,21 +138,43 @@ The Prometheus configuration now only needs to scrape from the OpenTelemetry Col
 
 Run the Flow and a sample request that we want to instrument locally. If the backends are running successfully the Flow has exported data to the Collector which can be queried and viewed.
 
+First start a Flow:
+
 ```python
-from jina import Flow, Document, DocumentArray
-import time
+from jina import Flow
 
 with Flow(
+    port=54321,
     tracing=True,
     traces_exporter_host='http://localhost',
     traces_exporter_port=4317,
     metrics=True,
     metrics_exporter_host='http://localhost',
     metrics_exporter_port=4317,
 ).add(uses='jinaai://jina-ai/SimpleIndexer') as f:
-    f.post('/', DocumentArray([Document(text='hello')]))
-    # allow some time for the metrics data export
-    time.sleep(3)
+    f.block()
+```
+
+Second execute requests using the instrumented {class}`jina.Client`:
+
+```python
+import time
+
+from jina import Client, Document, DocumentArray
+
+client = Client(
+    host='grpc://localhost:54321',
+    tracing=True,
+    traces_exporter_host='http://localhost',
+    traces_exporter_port=4317,
+)
+client.post('/', DocumentArray([Document(text='hello')]))
+# allow some time for the tracing data export
+time.sleep(3)
```

Comment:
no need to sleep here

Reply:
Just comment that there may be a delay

Reply:
Sleep is needed otherwise the Client process cannot export the data. You won't see the client span.

Reply:
Why would this be true? The Python process should not exit until the client has finished. This is how it should be.

If you have a client and u expect to see the logs, you should not aim to have an sleep, otherwise we have a huge bug here

Reply:
Its not about the client finishing but the usage of the batch exporter which is running in another thread. When the process exits it doesn't wait to flush the last batch. The other option is the expose the use the `client.teardown_instrumentation` method that I added that will ensure `flush` and `shutdown` calls.

Reply:
yes, I think this is what should happen. But I do not agree that the process does not wait for the threads to finish. Python will wait for them to finish AFAIK

Reply:
The thread will finish but the thread won't ensure that the last batch has been exported.